### PR TITLE
#33: add constraints verifications

### LINKED ISSUE #33: Add constraints verifications
This is to address the remaining points from the SC paper still not in code

### PR DESCRIPTION
**Pull request by ppebay**
Resolves #33

### CODE REVIEW

**Comment by ppebay** [comment]
Screenshot 2024-11-20 at 10 28 31

File: src/ccm_milp/generator.py [hunk 1]
```diff
@@ -457,14 +460,28 @@ def setup_milp(self):
...
+        if self.config.preserve_clusters:
+            n_constraints_theory += self.N
+        if n_constraints_equal == n_constraints_theory:
+            print(f"  {n_constraints_equal} equality constraints correctly added")
+        else:
+            print(f"*** ERROR: incorrect number of equality constrainsts added: {n_constraints_equal} <> {n_constraints_theory}")
+            sys.exit(1)
+
+        # Report on computed vs theoretical maximum number of inequality constraints were added
+        n_constraints_theory = self.I * ((self.K + 1) * self.N + 1)
```
Comment (pierrepebay) [inline src/ccm_milp/generator.py:477]: Where does this equation originate from in the paper?

Comment (ppebay) [reply src/ccm_milp/generator.py:477]: The most basic case (COMCP with no memory constraint) includes the following inequality constraints:
- the `I (K+1) N` integer shared block matrix relations (18) & (19)
- the `I` continuous maximum work relations (22)
thus yielding the absolute minimum of `I[(K+1) N + 1]` inequalities.

Comment (ppebay) [reply src/ccm_milp/generator.py:477]: Screenshot 2024-11-22 at 11 17 06

@pierrepebay

Comment (pierrepebay) [reply src/ccm_milp/generator.py:477]: thanks for the clarification

**Review by pierrepebay — APPROVED** [review]

**Pull request merged**